airflow/api_fastapi/core_api/routes/ui/assets.py

```diff
@@ -17,30 +17,36 @@
 
 from __future__ import annotations
 
-from fastapi import HTTPException, Request, status
+from fastapi import Depends, HTTPException, Request, status
 from sqlalchemy import and_, func, select
 
 from airflow.api_fastapi.common.db.common import SessionDep
 from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.security import requires_access_asset, requires_access_dag
 from airflow.models import DagModel
 from airflow.models.asset import AssetDagRunQueue, AssetEvent, AssetModel, DagScheduleAssetReference
 
 assets_router = AirflowRouter(tags=["Asset"])
 
 
-@assets_router.get("/next_run_assets/{dag_id}")
+@assets_router.get(
+    "/next_run_assets/{dag_id}",
+    dependencies=[Depends(requires_access_asset(method="GET")), Depends(requires_access_dag(method="GET"))],
+)
 def next_run_assets(
-    dag_id: str,
```

Comment:
and this should stay, sorry that was meant to be for the security dependency.

Reply:
Oh, yep, I think the failures were due to this removal. let me change it back